docs: adding documentation for paralus addon

niravparikh05 opened the pull request:
*Issue #, if available:*

*Description of changes:*

Adding documentation for paralus addon which got merged as part of https://github.com/aws-samples/cdk-eks-blueprints-patterns/pull/67

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

parkand1 reviewed (dismissed):
LGTM

shapirov103 reviewed (commented):
@niravparikh05 @parkand1 
If you read this doc, it is very confusing what pattern we are talking about, since the pattern is in another repo. My suggestion is this:

1. Change this PR and only provide a general description of the add-on and a link to the patterns repo with the paralus pattern implementation.
2. Move all pattern specific narrative away from this PR to a separate PR in the patterns repo.

Link both PRs. 

Please use existing OSS and commercial add-ons for reference. For example OSS add-on [pixie](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/docs/addons/pixie.md).

Or [DataDog](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/docs/addons/datadog.md) addon.

Just FYI, I am open to add a reference to Rafay in this PR to make it worthwhile their efforts. A small blurb on Rafay with a link to more advanced features for example, will be acceptable.

shapirov103 reviewed (changes requested):
See my comments

niravparikh05 commented:
@shapirov103 @parkand1 I have restructed the document addressing your comments and feedback, do have a look and let me know if any further changes are required.

shapirov103 commented on docs/addons/paralus.md:
```diff
@@ -0,0 +1,134 @@
+Paralus Addon deploys the following resources:
+
+- Creates EKS Cluster Control plane with public endpoint (for demo purpose only) with a managed node group
+- Deploys supporting add-ons:  AwsLoadBalancerController, VpcCni, KubeProxy, EbsCsiDriverAddOn
```
This can be mentioned for the add-on as "The add-on depends on the following add-ons: " - every customer will need to know that, regardless of whether they use the pattern or just need to use the add-on in their own blueprints.

niravparikh05 replied:
reframed as above

shapirov103 reviewed (commented):
Looks better for sure, please see a few comments I left. Happy to discuss.

parkand1 commented on docs/addons/paralus.md:
```diff
@@ -0,0 +1,134 @@
+
+The Paralus project is a free, open source tool that enables controlled, audited access to Kubernetes infrastructure. It comes with just-in-time service account creation and user-level credential management that integrates with your RBAC and SSO. [Learn more ..](https://www.paralus.io/)
+
+Paralus Blueprint Addon deploys paralus controller on your EKS cluster using [paralus construct](https://github.com/aws-samples/cdk-eks-blueprints-patterns/tree/main/lib/paralus-construct) implemented with the EKS Bleuprints [CDK](https://aws.amazon.com/cdk/). Detailed documentation on the same can be accessed from [here](https://github.com/aws-samples/cdk-eks-blueprints-patterns/blob/main/docs/p ...
```
Looks good to me here

niravparikh05 commented on docs/addons/paralus.md:
```diff
@@ -0,0 +1,134 @@
+Note: you can also refer to this [paralus eks blogpost](https://www.paralus.io/blog/eks-quickstart#configuring-dns-settings)
+
+## Disclaimer 
+This pattern relies on an open source NPM package paralus-eks-blueprints-addon. Please refer to the package npm site for more information.
```
removed this section

niravparikh05 reviewed (commented):
@shapirov103 have addressed your comments, please have a look

parkand1 reviewed (approved):
LGTM

shapirov103 reviewed (approved):
LGTM

@parkand1 there are some minor editorial changes that we will need to apply, such as rbac = RBAC, punctuation in some places
